explorer/explorerroutes.go

```diff
@@ -994,3 +994,59 @@ func (exp *explorerUI) AgendasPage(w http.ResponseWriter, r *http.Request) {
 	w.WriteHeader(http.StatusOK)
 	io.WriteString(w, str)
 }
+
+// StatsPage is the page handler for the "/stats" path
+func (exp *explorerUI) StatsPage(w http.ResponseWriter, r *http.Request) {
+	var stats StatsInfo
+	exp.MempoolData.RLock()
```

Comment:
Also need to RLock the NewBlockDataMtx, which guards both `ExtraInfo` and `NewBlockData`.

Reply:
@papacarp This was the thing I mentioned when talking about embedding the mutex in a struct.  It's error prone when it's not clear what the mutex goes with.